platforms/core-execution/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorProblemsApiIntegrationTest.groovy

```diff
@@ -163,39 +164,64 @@ class WorkerExecutorProblemsApiIntegrationTest extends AbstractIntegrationSpec {
             operationId == Long.parseLong(buildOperationIdFile.text)
             exception.message == "Exception message"
             exception.stacktrace.contains("Caused by: java.lang.Exception: Wrapped cause")
-            contextualLocations.size() == 1
-            (contextualLocations[0] as TaskLocation).buildTreePath == ":reportProblem"
-        }
-
-        def problem = Iterables.getOnlyElement(filteredProblemDetails(buildOperationsFixture))
-        with(problem) {
-            with(definition) {
-                name == 'type'
-                displayName == 'label'
-                with(group) {
-                    displayName == 'Generic'
-                    name == 'generic'
-                    parent == null
-                }
-                documentationLink == null
-            }
-            severity == Severity.WARNING.name()
+            definition.id.name == 'type'
+            definition.id.displayName == 'label'
+            definition.id.group.displayName == 'Generic'
+            definition.id.group.name == 'generic'
+            definition.id.group.parent == null
+            definition.documentationLink == null
+            definition.severity == Severity.WARNING
             contextualLabel == null
             solutions == []
             details == null
-            contextualLocations.empty
-            if (isolationMode == "'${WorkerExecutorFixture.IsolationMode.PROCESS_ISOLATION.method}'") {
-                // TODO: Should have the stack location for all isolation modes
-                assert originLocations.empty
-            } else {
-                assert originLocations.size() == 1
-                with(originLocations[0]) {
+            contextualLocations.size() == 1
+            with(contextualLocations[0] as TaskLocation) {
+                buildTreePath == ":reportProblem"
+                operationId == Long.parseLong(buildOperationIdFile.text)
+                exception.message == "Exception message"
```

Comment:
🤔  Is there an operationId and a exception on `TaskLocation`? Or is this a fall-through to the outer "with"?